src/commands/pipelines/DagRender.ts

```diff
@@ -0,0 +1,402 @@
+// Copyright(c) ZenML GmbH 2024. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0(the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.See the License for the specific language governing
+// permissions and limitations under the License.
+import fs from 'fs/promises';
+import * as vscode from 'vscode';
+import * as Dagre from 'dagre';
+import { ArrayXY, SVG, registerWindow } from '@svgdotjs/svg.js';
+import { PipelineTreeItem, ServerDataProvider } from '../../views/activityBar';
+import { PipelineRunDag, DagNode } from '../../types/PipelineTypes';
+import { LSClient } from '../../services/LSClient';
+import { ServerStatus } from '../../types/ServerInfoTypes';
+import { JsonObject } from '../../views/panel/panelView/PanelTreeItem';
+import { PanelDataProvider } from '../../views/panel/panelView/PanelDataProvider';
+
+const ROOT_PATH = ['resources', 'dag-view'];
+const CSS_FILE = 'dag.css';
+const JS_FILE = 'dag-packed.js';
+const ICONS_DIRECTORY = '/resources/dag-view/icons/';
+
+export default class DagRenderer {
+  private static instance: DagRenderer | undefined;
+  private openPanels: { [id: string]: vscode.WebviewPanel };
+  private createSVGWindow: Function = () => {};
+  private iconSvgs: { [name: string]: string } = {};
+  private root: vscode.Uri;
+  private javaScript: vscode.Uri;
+  private css: vscode.Uri;
+
+  constructor(context: vscode.ExtensionContext) {
+    DagRenderer.instance = this;
+    this.openPanels = {};
+    this.root = vscode.Uri.joinPath(context.extensionUri, ...ROOT_PATH);
+    this.javaScript = vscode.Uri.joinPath(this.root, JS_FILE);
+    this.css = vscode.Uri.joinPath(this.root, CSS_FILE);
+
+    this.loadSvgWindowLib();
+    this.loadIcons(context.extensionPath + ICONS_DIRECTORY);
+  }
```

Comment:
The issue is that the typing of `createSVGWindow` is generic specifically because it will hold a dynamically imported function from `svgdom`. The initial value is just a placeholder and will have a different signature once the dynamic import takes place. `svgdom` is not compatible with the standard way to import it with our TypeScript settings and we can not pull types from it until it is dynamically imported.